src/zarr/core/group.py

```diff
@@ -673,6 +674,11 @@ def from_dict(
         store_path: StorePath,
         data: dict[str, Any],
     ) -> AsyncGroup:
+        node_type = data.pop("node_type", None)
+        if node_type == "array":
+            raise ContainsArrayError(store_path.store, store_path.path)
```

Comment:
i recommend specifying the exception message explicitly as a single argument instead of relying on the automatic formatting.

Reply:
👍 I added custom messages